keps/sig-cli/NNNN-kustomize-components.md

```diff
@@ -0,0 +1,308 @@
+> required) or even code snippets.  If there's any ambiguity about HOW your
+> proposal will be implemented, this is the place to discuss them.
+
+
+### Test Plan
+
+(**FIXME:** Mention that we will add unit tests)
+
+> **Note:** *Not required until targeted at a release.*
+>
+> Consider the following in developing a test plan for this enhancement:
+> - Will there be e2e and integration tests, in addition to unit tests?
+> - How will it be tested in isolation vs with other components?
+>
+> No need to outline all of the test cases, just the general strategy.  Anything
+> that would count as tricky in the implementation and anything particularly
+> challenging to test should be called out.
+>
+> All code is expected to have adequate tests (eventually with coverage
+> expectations).  Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+> when drafting this test plan.
+>
+> [testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+
+### Graduation Criteria
+
+(**FIXME:** Maybe have two major releases of Kustomize for alpha -> beta, write
+user-facing documentation, etc. See
+[here](https://github.com/arrikto/kubernetes-enhancements/blob/master/keps/sig-cli/20200115-kubectl-diff.md#graduation-criteria)
+for an example)
+
+> **Note:** *Not required until targeted at a release.*
+>
+> Define graduation milestones.
+>
+> These may be defined in terms of API maturity, or as something else. The KEP
+> should keep this high-level with a focus on what signals will be looked at to
+> determine graduation.
+>
+> Consider the following in developing the graduation criteria for this enhancement:
+> - [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+> - [Deprecation policy][deprecation-policy]
+>
+> Clearly define what graduation means by either linking to the [API doc
+> definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+> or by redefining what graduation means.
+>
+> In general, we try to use the same stages (alpha, beta, GA), regardless how the
+> functionality is accessed.
+>
+> [maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+> [deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+## Implementation History
+
+(**FIXME:** This section should be left empty, but not be removed)
+
+> Major milestones in the life cycle of a KEP should be tracked in this section.
+> Major milestones might include
+> - the `Summary` and `Motivation` sections being merged signaling SIG acceptance
+> - the `Proposal` section being merged signaling agreement on a proposed design
+> - the date implementation started
+> - the first Kubernetes release where an initial version of the KEP was available
+> - the version of Kubernetes where the KEP graduated to general availability
+> - when the KEP was retired or superseded
+
+## Drawbacks
```

Comment:
I think that the **Drawbacks** section can be removed. I can't think of any, and most of the Kustomize KEPs leave this section out.